### ISSUE TITLE
autoexec section does not work

### ISSUE THREAD

**Issue by dreamer** [flathub/com.dosbox.DOSBox#20]
On up-to-date Fedora 30:

Having file `foo.conf` in home directory, with content:

    [autoexec]
    echo baz

Expected behaviour:

    $ dosbox -c 'echo bar'
    # in DOSBox window:
    Z:\echo bar
    bar
    $ dosbox -conf foo.conf
    # in DOSBox window, autoexec section is injected:
    Z:\echo baz
    baz

Actual behaviour:

    $ flatpak run com.dosbox.DOSBox -c 'echo bar'
    # in DOSBox window:
    Z:\echo bar
    bar
    $ flatpak run com.dosbox.DOSBox -conf foo.conf
    # in DOSBox window, autoexec section is not injected:
    Z:\

This makes DOSBox installed through flatpak unusable with [steam-dos](https://github.com/dreamer/steam-dos).

**Comment by AsciiWolf** [comment]
Yep, this is intended behavior - DOSBox is sandboxed and has access only to the Documents directory (xdg-documents) by default. You can however change this using [flatpak override](http://docs.flatpak.org/en/latest/flatpak-command-reference.html#flatpak-override).

**Comment by dreamer** [comment]
I'm sorry, but this makes DOSBox flatpak package unusable... Almost 100% of software distributed for DOSBox (through Steam, GOG or e.g. Lutris) comes preconfigured using autoexec section...

**Comment by AsciiWolf** [comment]
It will work if the software and autoexec file are stored in Documents directory. I don't want to give the DOSBox Flatpak full home permissions, sorry.

**Comment by dreamer** [comment]
Hmm, I understand. 

It's somewhat unfortunate, as dosbox flatpak not working out of the box was the only thing that prevented me from recommending this package to the users - as this is the first package to release 0.74-3, which provides some essential fixes for my project. For now I documented it on [wiki page](https://github.com/dreamer/steam-dos/wiki/Flatpak).

The deal is - for my project, DOSBox needs read access to the Steam game library to read actual game files (by default, it's in `~/.local/share/Steam`, and I presume write access would be needed for save games and such). Some DOS games might be as large as 1-1.5 GB due to CD images included, so simply copying them to ~/Documents is out of the question.

As far a configuration files are concerned, I am generating them in game's installation dir - but can place them anywhere, really - including some location, that would be whitelisted by Flatpak.

Maybe we can find some arrangement that would provide out-of-the-box functionality for end users?

**Comment by TingPing** [comment]
`--filesystem=xdg-data/Steam:ro` is probably harmless (or does it need `:create`?)

**Comment by TingPing** [comment]
> $ sudo flatpak override --filesystem=home com.dosbox.DOSBox


That documentation is very broken though. This package uses `--persist` which is in conflict with `home` permissions.

**Comment by dreamer** [comment]
@TingPing 

> --filesystem=xdg-data/Steam:ro

I don't think it's enough. I am sure various games will expect to write and overwrite save files, configuration files, etc.

> That documentation is very broken though. This package uses --persist which is in conflict with home permissions.

Is it? manual for flatpak-override says:

    --persist=FILENAME
        If the application doesn't have access to the real homedir (…)

So read access to home seems to override persist option?

Anyway - if it's possible to get rid of that override and still have working DOSBox, then I'm all for it! Feel free to edit that Wiki page - it's open for everyone to edit.

If it was possible to sandbox dosbox access to game's installation directory (which might be even outside of home, mind you!) using only parameters to `flatpak run`, I would be all for it!

Would something like `flatpak run --filesystem=<absolute path to game's installation dir> com.dosbox.DOSBox` work with this flatpak package?

**Comment by TingPing** [comment]
> So read access to home seems to override persist option?

Yes because the real home dir is mounted. So running that command just results in users losing old configs.

> Would something like flatpak run --filesystem=<absolute path to game's installation dir> com.dosbox.DOSBox work with this flatpak package?

Yes that is entirely valid.

**Comment by dreamer** [comment]
OK, so I implemented detection of game installation directory and instructed users to use this handy and completely intuitive command if they want to use flatpakked dosbox:

    flatpak run --filesystem=%install_dir% com.dosbox.DOSBox

I understand now why there is no read access to home by default...

But do you really think it's OK to fail without leaving proper information to the end user? Shouldn't flatpak leave some warning on stderr when DOSBox reaches outside of it's sandbox instead of quietly failing?

If you think it's OK and expected behaviour, I will simply close this issue.

**Comment by AsciiWolf** [comment]
@dreamer It's clearly mentioned in the Flatpak AppData: `This version of DOSBox is sandboxed. You have to copy your DOS files to your Documents directory and mount them using a mount command in DOSBox.` :-)

**Comment by dreamer** [comment]
Hmm, I found package using `flatpak search dosbox` and installed with `flatpak install com.dosbox.DOSBox` (I avoid Gnome Software for multiple reasons). Just repeated the process to see if I missed it and turns out I did:

    com.dosbox.DOSBox permissions:
        ipc      network      pulseaudio      x11     devices     file access [1]

        [1] xdg-documents

Well, not the most user-friendly way to inform user that application is sandboxed outside of Documents dir, but I can't argue - it's there. At least I learned something ;)

I guess further improvements will require changes to the flatpak itself and should not be directed at random package maintainers - thank you for your help! :)
